packages/manager/apps/pci-kubernetes/src/pages/detail/restrictions/Restrictions.page.tsx

```diff
@@ -1,3 +1,188 @@
+import { ResponseAPIError } from '@ovh-ux/manager-pci-common';
+import {
+  Datagrid,
+  Notifications,
+  useDataGrid,
+  useNotifications,
+} from '@ovhcloud/manager-components';
+import { ODS_THEME_COLOR_INTENT } from '@ovhcloud/ods-common-theming';
+import {
+  ODS_BUTTON_SIZE,
+  ODS_BUTTON_VARIANT,
+  ODS_ICON_NAME,
+  ODS_ICON_SIZE,
+  ODS_SPINNER_SIZE,
+  ODS_TEXT_LEVEL,
+  ODS_TEXT_SIZE,
+} from '@ovhcloud/ods-components';
+import {
+  OsdsButton,
+  OsdsIcon,
+  OsdsSpinner,
+  OsdsText,
+} from '@ovhcloud/ods-components/react';
+import { Translation, useTranslation } from 'react-i18next';
+import { useParams } from 'react-router-dom';
+import { formatIP } from '@/helpers';
+import {
+  useDeleteRestriction,
+  useMappedRestrictions,
+  useUpdateRestriction,
+} from '@/api/hooks/useRestriction';
+import { useRestrictionColumns } from '../../../components/restriction/useRestrictionColumns';
+
 export default function RestrictionsPage() {
-  return <>Kube restrictions</>;
+  const { t } = useTranslation('restrictions');
+  const { kubeId, projectId } = useParams();
+  const { pagination, setPagination } = useDataGrid();
+  const { addError, addSuccess } = useNotifications();
+
+  const {
+    data,
+    isPending: isFetchPending,
+    addEmptyRow,
+    deleteRowByIndex,
+  } = useMappedRestrictions(projectId, kubeId, {
+    pagination,
+  });
```

Comment:
since ```{ pagination }``` is a new object reference at each render you probably just want to pass ```pagination``` instead otherwise the useEffect change i suggested above would be triggered at each render